### DISCUSSION TITLE
JSONRPC Schemas

### DISCUSSION THREAD

**Discussion by Jack-Above** [tendermint/tendermint#9633] [Q&A]
Hello all,

I was looking for JSONRPC response schemas and found them in yaml in [openapi.yaml](https://github.com/tendermint/tendermint/blob/1d25a3f0bc7994aa12c48cf600bb756ecbb4756d/rpc/openapi/openapi.yaml#:~:text=components%3A-,schemas,-%3A).  However, these appear to be incorrect. For instance, on `BroadcastTxResponse`, both `result` and `error` are required, but the API returns one or the other exclusively.  Also, `error` is declared as `type: string`, but an object is returned in testing. 

We are using v0.34.16; anything I am missing here? Thanks in advance.

**Comment by thanethomson** [comment]
You are correct - the OpenAPI spec is incorrect in a couple of places.

I think the original authors meant that either `result` _or_ `error` are required. I'm not sure how to represent that according to the OpenAPI spec. I assume we could simply just make both of them not required?

With regard to the `ErrorResponse` type, while the OpenAPI spec claims that the error is a string, this is actually the object that is returned: https://github.com/tendermint/tendermint/blob/cceea4de2221b0f2b490247fa5619edf6b1048ee/rpc/jsonrpc/types/types.go#L154-L159

And the error part is actually this object: https://github.com/tendermint/tendermint/blob/cceea4de2221b0f2b490247fa5619edf6b1048ee/rpc/jsonrpc/types/types.go#L140-L144

I'll convert this discussion into an issue so we can fix this.
